modules/renter/proto/contract.go

```diff
@@ -411,17 +399,17 @@ func (cs *ContractSet) managedInsertContract(h contractHeader, roots []crypto.Ha
 	if err != nil {
 		return modules.RenterContract{}, err
 	}
-	// preallocate space for header + roots
-	if err := f.Truncate(contractHeaderSize + crypto.HashSize*int64(len(roots))); err != nil {
-		return modules.RenterContract{}, err
-	}
+	// create fileSections
+	headerSection := newFileSection(f, 0, contractHeaderSize)
+	rootsSection := newFileSection(f, contractHeaderSize+1, -1)
```

Comment:
my preference would be to change the `-1` sentinel to a constant. It's not super obvious, had to read the implementation to figure out what it did. But if it was a constant `remainingFile`, it might be more clear.